### ISSUE TITLE
Single-File Release version of header?

### ISSUE THREAD

**Issue by bryceschober** [andrew-gresyk/HFSM#13]
It would be nice if there was a "pre-compiled" version of the library into a single header file (with license included at the top). This would enable playing around with examples of using the library in an online sandbox (other than compiler explorer, which I see can include HFSM). Wandbox lets you create as many headers as you like, but building the complete set for HFSM would be tedious. It does [include](https://github.com/melpon/wandbox-builder/blob/master/build-head.sh) [some](https://github.com/melpon/wandbox-builder/blob/master/cattleshed-conf/compilers.py) header-only libs, but not yet HFSM, and that's not a general solution for the many other useful online sandboxes. The current many-file version is good for maintenance and implementation-hiding, but prohibits these other uses.

In short, could you make an example like [this wandbox session](https://wandbox.org/permlink/9GmHezj8IjuRkO8S) "just work"?

FYI, it seems that the [ned14/pcpp](https://github.com/ned14/pcpp) project is intended for this use case, and one example usage can be found [here](https://github.com/psteinb/compass/blob/master/CMakeLists.txt#L95).

**Comment by andrew-gresyk** [comment]
Yeah, sounds useful.
I'll see if I can add HFSM to wandbox.

**Comment by bryceschober** [comment]
So... I actually tried the [streetlight example in Compiler Explorer](https://godbolt.org/g/G9KxPE) and it wasn't able to find the HSFM header... Does it show up in a different include path? How is it *supposed* to work after selecting the library?

**Comment by andrew-gresyk** [comment]
Trunk seems broken, I'll have look.
Meanwhile, try switching to 0.8.

**Comment by bryceschober** [comment]
OK, now with 0.8, x86 gcc 5.x works, but arm-none gcc 5.4.1 has a problem finding <memory.h>, and arm-linux 5.4 has a static assert error...

**Comment by bryceschober** [comment]
I suspect the <memory.h> problem could be circumvented by including <cstring> instead.

**Comment by andrew-gresyk** [comment]
Added https://github.com/andrew-gresyk/HFSM/blob/master/include/hfsm/machine_single.hpp
Also, removed <memory.h> include.
